### PR TITLE
Define lockfile v1 schema

### LINKED ISSUE #11: recovery: Define lockfile v1 schema
## Context

rpm.lock is the source of reproducibility. It must record resolved facts, not just requested dependency ranges. Later installer and semver work depends on this contract being clear early.

## Contract

- Lockfile entries record package name, requested range, resolved version, tarball URL, integrity or shasum, and dependency edges.
- The schema distinguishes direct, dev, and transitive relationships where needed.
- Empty lockfile handling is safe and deterministic.
- Parse and save errors include path and useful context.

## Done criteria

- Empty lockfile initializes safely.
- Lockfile records resolved versions, not only requested ranges.
- Save operation truncates old content correctly.
- Parse errors include file path and useful context.

### PR DESCRIPTION
**Pull request by nerdchanii**
## Contract

Defines the `rpm.lock` v1 contract for reproducible installs. This is contract-affecting because it covers lockfile fields, compatibility, safe empty-lock handling, save behavior, manifest range preservation, and parse/save diagnostics.

## Changes

- Added `docs/specs/lockfile-v1.md` as the owning lockfile contract.
- Updated lockfile entries to carry package name, requested range, resolved version, relationship, tarball URL, integrity/shasum, dependency edges, and project metadata.
- Made empty lockfiles initialize as an empty v1 lockfile for mutation, while non-mutating node_modules linking rejects empty package plans before replacing an existing tree.
- Made non-empty lockfiles missing `lockfile_version = 1` or declaring unsupported versions fail with path-aware errors.
- Persisted selected registry dist metadata and direct/dev/transitive relationships from `rpm add`/`rpm install` into lockfile entries without demoting root dependencies or replacing root requested ranges during merges.
- Preserved manifest dependency ranges in `requested` and direct `package.json` updates while preserving the legacy normalized registry request for current resolver behavior.
- Added lockfile/package manifest/node linker/add fixtures/tests for v1 fields, project metadata, empty load, missing/unsupported versions, path-aware parse errors, relationship/requested merge behavior, range preservation, empty-link rejection, registry request normalization, and save truncation.

## Validation

- `cargo fmt --check` passed.
- `cargo test lock_file_test` passed.
- `cargo test package_json_test` passed.
- `cargo test node_linker` passed.
- `cargo test registry_request_preserves_legacy_manifest_range_normalization` passed.
- `cargo test manifest_version_preserves_requested_range_for_direct_adds` passed.
- `cargo check` passed.
- `cargo test` passed. The expected shell message for the missing-binary run test was printed. One earlier overlapped validation run raced with a simultaneous targeted cargo test, so the full suite was rerun by itself and passed.
- `cargo clippy --all-targets --all-features` passed with existing warnings in `lockfile/constraint.rs`, `package_manifest`, `api`, and older `registry` code.

## Checklist

- [x] Scope is focused on one behavior, bug, or mechanical change.
- [x] Behavior changes include relevant tests or fixtures.
- [x] SPEC impact is classified when contract-affecting.
- [ ] PR is ready for review when implementation is complete.

Closes #11

### CODE REVIEW

**Comment by nerdchanii** [comment]
@codex review

**Comment by nerdchanii** [comment]
@codex review

**Comment by nerdchanii** [comment]
@codex review

**Comment by nerdchanii** [comment]
@codex review

**Comment by nerdchanii** [comment]
@codex review

**Comment by nerdchanii** [comment]
@codex review

**Comment by nerdchanii** [comment]
@codex review

**Comment by nerdchanii** [comment]
@codex review

File: src/lib/command/working_process/add.rs [hunk 1]
```diff
@@ -10,40 +15,103 @@ pub async fn add(
...
+    let mut version = requested.to_string();
+    if version.contains("||") {
+        version = version
+            .split("||")
+            .last()
+            .map(|version| version.trim().to_string())
+            .unwrap_or_default();
+    }
+    if version.starts_with('^') || version.starts_with('~') {
+        version.remove(0);
```
[the remark this answers is not in the document]
Comment (nerdchanii) [reply src/lib/command/working_process/add.rs:100]: Rejecting this one for #11 scope. This PR keeps the existing registry request normalization isolated in add.rs so the new lockfile schema can preserve the original requested range while remaining compatible with the current resolver behavior. Semver-aware range selection is the resolver contract work tracked separately in docs/specs/resolver.md and issue #1; pulling that resolver change into this lockfile schema PR would cross the intended boundary.

**Comment by nerdchanii** [comment]
@codex review

File: src/lib/command/working_process/add.rs [hunk 2]
```diff
@@ -10,40 +15,124 @@ pub async fn add(
...
```
[the remark this answers is not in the document]
Comment (nerdchanii) [reply src/lib/command/working_process/add.rs:100]: Accepted as a real resolver issue, but not as a PR #37 lockfile-schema fix. PR #37 intentionally preserved current range-normalization behavior while adding `requested` and resolved `version` lockfile fields. Semver-aware highest-matching selection is the scope of issue #1 and is now being handled in PR #40 with an authoritative semver-resolution contract, fixture cases, and replacement targets for this helper. Leaving this linked here for traceability rather than changing merged lockfile-schema code retroactively.
